src/main/java/org/jboss/ejb/client/EJBClientContext.java

```diff
@@ -330,6 +335,82 @@ static InterceptorList getClassPathInterceptorList() {
     }
 
     /**
+     * Register a client interceptor with this client context.
+     * <p/>
+     * If the passed <code>clientInterceptor</code> is already added to this context with the same <code>priority</code>
+     * then this method just returns the old {@link org.jboss.ejb.client.EJBClientInterceptor.Registration}. If however,
+     * the <code>clientInterceptor</code> is already registered in this context with a different priority then this method
+     * throws an {@link IllegalArgumentException}
+     *
+     * @param priority          the absolute priority of this interceptor (lower runs earlier; higher runs later)
+     * @param clientInterceptor the interceptor to register
+     * @return a handle which may be used to later remove this registration
+     * @throws IllegalArgumentException if the given interceptor is {@code null}, the priority is less than 0, or the
+     *                                  given interceptor is already registered with a different priority
+     * @deprecated
```

Comment:
An explanation for the deprecation should be made here.  Also, as a matter of convention I also add the ``@Deprecated`` annotation.

Reply:
There is also a problem that if an interceptor is added or removed after a proxy is used, the proxy will not see the update because proxy interceptor information is cached by way of the ``org.jboss.ejb.client.EJBClientContext#proxyInfoValue`` field.  This might be OK (since this is just a compatibility method) but should be documented in any event.